### ISSUE TITLE
On Magento 2, problem when using custom order numbering extension and paypal

### ISSUE THREAD

**Issue by ipekerti** [eltrino/PrintOrder#3]
We were trying to use this extension on M2 Community (we're now on v2.1), and custom order numbering extension from Amasty.

> When PayPal is used as payment method, the order # is not being recognized once the control is handed back to Magento from PayPal.
> - Order placed
> - User sent to PayPal
> - User completes payment process on PayPal
> - User is handed back to Magento
>   -- This is the point where Magento is not recoginizing the custom order #

Removed the Eltrino PrintOrder extension, PayPal checkout working again.

**Comment by odi-um** [comment]
Hi @ipekerti, 

Do you have any additional data like logs ? Have you turned on php error reporting to see errors ? 

On our side we will check extension with latest 2.1 release.

**Comment by ipekerti** [comment]
No, we don’t have any logs, we ran into this problem when we were on v2.0, we had since removed the extension and upgraded to v2.1.

Am just reporting the problem when I happened to have stumbled onto the github issue.

We believe this is what’s happening:
- ```
       We’re using a custom order numbering extension (i.e. W1000393, instead of the internal M2 order ID # 232), normally the link generated for an end-of-order print confirmation (using a non-PayPal payment method) is able to map to that custom order #.  https://m2dev.specialized.net/sales/order/view/order_id/232/
  ```
- ```
       Since for non-PayPal payment method the session never has to leave Magento’s site, it seems to be able to retain the custom order # association to Magento’s internal order ID #.
  ```

[cid:image001.png@01D2051F.2CCDDE20]

However, when PayPal is used as payment method guest user:
- Order placed
- User sent to PayPal
- User completes payment process on PayPal’s site
- When control is handed back to Magento’s site, we think somehow the mapping of Magento’s internal order # and the custom order # is lost in the process.

Thanks!
-Immanuel Pekerti
